src/registrar/management/commands/utility/terminal_helper.py

```diff
@@ -75,28 +76,74 @@ class PopulateScriptTemplate(ABC):
     run_summary_header = None
 
     @abstractmethod
-    def update_record(self, record):
-        """Defines how we update each field. Must be defined before using mass_update_records."""
+    def update_record(self, record: Model):
+        """Defines how we update each field.
+
+        raises:
+            NotImplementedError: If not defined before calling mass_update_records.
+        """
         raise NotImplementedError
 
-    def mass_update_records(self, object_class, filter_conditions, fields_to_update, debug=True):
+    def mass_update_records(self, object_class: Model, filter_conditions, fields_to_update, debug=True, verbose=False, custom_filter=None):
         """Loops through each valid "object_class" object - specified by filter_conditions - and
         updates fields defined by fields_to_update using update_record.
 
-        You must define update_record before you can use this function.
+        Parameters:
+            object_class: The Django model class that you want to perform the bulk update on.
+                This should be the actual class, not a string of the class name.
+
+            filter_conditions: dictionary of valid Django Queryset filter conditions 
+                (e.g. {'verification_type__isnull'=True}).
+
+            fields_to_update: List of strings specifying which fields to update.
+                (e.g. ["first_ready_date", "last_submitted_date"])
+
+            debug: Whether to log script run summary in debug mode. 
+                Default: True.
+
+            verbose: Whether to print a detailed run summary *before* run confirmation. 
+                Default: False.
+
+            custom_filter: function taking in a single record and returning a boolean representing whether
+                the record should be included of the final record set. Used to allow for filters that can't be
+                represented by django queryset field lookups. Applied *after* filter_conditions.
+
+        Raises:
+            NotImplementedError: If you do not define update_record before using this function.
+            TypeError: If custom_filter is not Callable.
         """
 
         records = object_class.objects.filter(**filter_conditions)
+
+        # apply custom filter
+        if custom_filter:
```

Comment:
I decided to give this custom_filter logic a shot instead of using should_skip_record because the latter was described in print statements as being for records that need to be updated manually, whereas what I really wanted was to add a filter that I couldn't express using Djangos regular field lookups. Very open to changing this.